aiohttp/web_request.py

```diff
@@ -510,8 +510,26 @@ def text(self):
         return bytes_body.decode(encoding)
 
     @asyncio.coroutine
-    def json(self, *, loads=json.loads):
-        """Return BODY as JSON."""
+    def json(self, *, loads=json.loads, allowed_types=None, err_handler=None):
```

Comment:
Shall we directly check if the mime type starts with `'application/'` and ends with `'+json'` i.e. [flask's way to deal with it](https://github.com/pallets/flask/blob/0.12-maintenance/flask/wrappers.py#L119)?  Hence no need to consider about how should allowed_type behavior